Project.swift

```diff
@@ -552,6 +552,14 @@ func targets() -> [Target] {
 
 let project = Project(
     name: "Tuist",
+    options: [
+        .textSettings(
```

Comment:
I wonder if it wouldn't make sense to use the enum with associated values rather than having nested `.textSettings(.textSettings(...))`

You could have something like:
```swift
public enum ProjectOption: Codable, Equatable {
    /// Text settings to override user ones for current project
    case textSettings(
        usesTabs: Bool? = nil,
        indentWidth: UInt? = nil,
        tabWidth: UInt? = nil,
        wrapsLines: Bool? = nil
    )
}
```
We'll probably add more enums to `options`, so I think it's important to get this right - it will also look a bit awkward if we have the same duplication of the same name for other flags as well

Reply:
I actually wanted to add the same comment but it get lost somewhere 🤷‍♂️ 
so +1 for Marek's proposal 🙌

Reply:
👍